sklearn/datasets/_svmlight_format.pyx

```diff
@@ -61,7 +61,7 @@ def _load_svmlight_file(f, dtype, bint multilabel, bint zero_based,
     for line in f:
         # skip comments
         line_cstr = line
-        hash_ptr = strchr(line_cstr, '#')
+        hash_ptr = strchr(line_cstr, 35)  # ASCII value of '#' is 35
```

Comment:
Nice! This was the part missing in https://github.com/scikit-learn/scikit-learn/pull/12873 that prevented me to use a global option.